client/my-sites/checkout/composite-checkout/components/wp-checkout-order-review.tsx

```diff
@@ -133,6 +168,7 @@ export default function WPCheckoutOrderReview( {
 	};
 
 	const planIsP2Plus = hasP2PlusPlan( responseCart );
+	const shouldShowDomainNote = experiment?.variationName && domainUrl?.includes( '.wordpress.com' );
```

Comment:
Just wanted to give you a heads up that `.includes( '.wordpress.com' )` will return `true` also for `x.wordpress.comaliciousdomain.biz` as well, and we might want to make it a bit more precise. 

Perhaps if we use `String.prototype.endsWith()`?